pkg/kubectl/cmd/resource/get.go

```diff
@@ -284,6 +301,15 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 
 	objs := make([]runtime.Object, len(infos))
 	for ix := range infos {
+		if options.ServerPrint {
+			table, err := options.decodeIntoTable(f.JSONEncoder(), infos[ix].Object)
+			if err == nil {
```

Comment:
Maybe add a debug statement at level 2 with the error and info that you're falling back to generic. It might save us some debugging cycles.